### PR TITLE
Seeding

### PR DESCRIPTION
**Pull request by javsflores**
@bland0 @jase-d 

Hey guys,

this commit has 1) a couple of edits to my database schema, plus a restructuring of the functions per Jase's advise to try to separate concerns. (I think I did what Jase recommended). 

2) My complete seed.js file - the file looks very long but the code should be relatively simple and easy to follow.

Thanks in advance to whoever reviews it.

### CODE REVIEW

File: database/seed.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  var avgAcc = Math.floor((Math.random() * 5) + 1);
+  var avgComm = Math.floor((Math.random() * 5) + 1);
+  var avgLoc = Math.floor((Math.random() * 5) + 1);
+  var avgCheckIn = Math.floor((Math.random() * 5) + 1);
+  var avgVal = Math.floor((Math.random() * 5) + 1);
+  var totalAvg = (avgCln+avgAcc+avgComm+avgLoc+avgCheckIn+avgVal)/6
+  var hostID = Math.floor((Math.random() * 500) + 1);
+
+  var sql = `INSERT INTO reviews_locations (location_id, location_name, location_hostID, location_avg, location_avgCln, location_avgAcc, location_avgComm, location_avgLoc, location_avgCheckIn, location_avgVal)
+  VALUES (${i} , "${faker.address.streetName()}", ${hostID}, ${totalAvg}, ${avgCln}, ${avgAcc}, ${avgComm}, ${avgLoc}, ${avgCheckIn}, ${avgVal});`;
```
Comment (bland0) [inline database/seed.js:64]: Be sure that streetName() never contains apostrophes.

File: database/index.js [hunk 2]
```diff
@@ -21,85 +21,77 @@ db.query(`CREATE DATABASE IF NOT EXISTS ${database}`, (err, result) => {
...
-    db.query(hostsTable, (err, result) => {
-      if (err) {
-        throw err;
-      }
-      console.log('Hosts table has been created!')
-    })
+const usersTable = `CREATE TABLE IF NOT EXISTS reviews_users (
+  user_id INT AUTO_INCREMENT PRIMARY KEY,
+  user_name VARCHAR(255),
+  user_img VARCHAR(255)
```
Comment (bland0) [inline database/index.js:37]: Will the URL ever be longer than 255 bytes?

File: database/index.js [hunk 3]
```diff
@@ -21,85 +21,77 @@ db.query(`CREATE DATABASE IF NOT EXISTS ${database}`, (err, result) => {
...
-      host_img INT
-    )`
-    db.query(hostsTable, (err, result) => {
-      if (err) {
-        throw err;
-      }
-      console.log('Hosts table has been created!')
-    })
+const usersTable = `CREATE TABLE IF NOT EXISTS reviews_users (
+  user_id INT AUTO_INCREMENT PRIMARY KEY,
```
Comment (bland0) [inline database/index.js:35]: What's the purpose of AUTO_INCREMENT when you're assigning user_id manually? If that's the case, could you use zero-based indexing if you nixed AUTO_INCREMENT?

File: database/seed.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  })
+}
+
+for (var i = 0; i < 5001; i++) {
+
+  var locationID = Math.floor((Math.random() * 100) + 1);
+  var userID = Math.floor((Math.random() * 3000) + 1);
+
+  var sql = `INSERT INTO reviews (review_id, review_msg, review_locationID, review_userID, review_createdAt)
+  VALUES (${i} , "${faker.lorem.sentences()}", ${locationID}, ${userID}, "${new Date}");`;
```
Comment (bland0) [inline database/seed.js:82]: Be sure that sentences() never contains apostrophes.

**Review by bland0 — APPROVED** [review]
A few potential issues with queries

**Comment by javsflores** [comment]
@bland0 thanks for the great feedback!

- for the street names, I'm hoping they do not contain apostrophes, but I also think that by using the double quotes, I'm taking care of that issue. I had a similar issue before with a string that had an apostrophe and I had wrapped around single quotes and it was a mess to figure out lol.

- In regards of the URL, I'm hoping it isn't longer than 255 characters, that would be quite the long url, but maybe I can convert that to TEXT instead of VARCHAR. What do you think of that? Would there be any possible issues that might arise from converting it to TEXT data type?

- very good point on the auto increment comment, I honestly didn't even think of that for some reason. I'll go ahead and adjust that, thanks!

- yeah, apostrophes are the worse lol.

Thanks, Branan, I appreciate the feedback!

**Comment by bland0** [comment]
@javsflores
The second argument V of `db.query(Q, V, F)` is an array that takes placeholder values.
[https://www.npmjs.com/package/mysql#performing-queries](https://www.npmjs.com/package/mysql#performing-queries)

**Comment by javsflores** [comment]
Hey @bland0,

I implemented the suggestions you made!

Thanks again,
Javier

**Pull request merged**